app_utils/shared_utils.py

```diff
@@ -127,7 +123,7 @@ def get_snowflake_connection() -> SnowflakeConnection:
                 return get_connector().open_connection(db_name="")
 
 
-@st.cache_resource(show_spinner=False)
+# @st.cache_resource(show_spinner=False)
```

Comment:
Sorry, I forgot to comment this back. That said, I didn't understand why we needed this. AND in debugging a live app, I had issues with this cached, where it would say `session_state` doesn't exist sometimes.

Reply:
I think its cached to not create a new session object each time.